apps/files_sharing/lib/Controller/ShareesAPIController.php

```diff
@@ -197,6 +197,27 @@ public function search(string $search = '', ?string $itemType = null, int $page
 			$result['exact'] = array_merge($this->result['exact'], $result['exact']);
 		}
 		$this->result = array_merge($this->result, $result);
+
+		// Get a list of groups to exclude from search results
+		$blacklisted_groups = $this->config->getSystemValue('blacklisted_groups', true);
+
+		// Check and filter normal 'groups' array at the top level and the nested 'exact' level
+		if (isset($this->result['groups'])) {
+		    $this->result['groups'] = array_filter($this->result['groups'], function($group) use ($blacklisted_groups) {
+		        return !in_array($group['label'], $blacklisted_groups);
+		    });
+		    // Reindex array to maintain sequential numeric keys
+		    $this->result['groups'] = array_values($this->result['groups']);
+		}
+
+		if (isset($this->result['exact']['groups'])) {
+		    $this->result['exact']['groups'] = array_filter($this->result['exact']['groups'], function($group) use ($blacklisted_groups) {
+		        return !in_array($group['label'], $blacklisted_groups);
+		    });
+		    // Reindex array to maintain sequential numeric keys
+		    $this->result['exact']['groups'] = array_values($this->result['exact']['groups']);
+		}
```

Comment:
```suggestion
       $filterBlacklistedGroups = function ($groups) use ($blacklisted_groups) {
            return array_values(array_filter($groups, function ($group) use ($blacklisted_groups) {
                return isset($group['label']) && !in_array($group['label'], $blacklisted_groups);
            }));
        };

        if (isset($this->result['groups'])) {
            $this->result['groups'] = $filterBlacklistedGroups($this->result['groups']);
        }

        if (isset($this->result['exact']['groups'])) {
            $this->result['exact']['groups'] = $filterBlacklistedGroups($this->result['exact']['groups']);
        }
```

Since the filter logic is repeated, you can use an inline closure..... of define a function outside this method.